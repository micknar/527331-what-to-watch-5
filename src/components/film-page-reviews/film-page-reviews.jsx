import React, {useEffect} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {fetchComments} from "../../store/api-actions";
import {shadeColor} from "../../utils";

const FilmPageReviews = (props) => {
  const {comments, filmId, loadComments, backgroundColor} = props;

  useEffect(() => {
    loadComments(filmId);
  }, [filmId]);

  return (
    <div className="movie-card__reviews movie-card__row">
      <div className="movie-card__reviews-col">
        {comments.map((comment, i) => {
          return (
            <div
              className="review"
              key={`${i}-${comment.user.id}`}
              style={{borderBottomColor: shadeColor(backgroundColor, -20)}}
            >
              <blockquote className="review__quote">
                <p className="review__text">{comment.comment}</p>

                <footer className="review__details">
                  <cite className="review__author">{comment.user.name}</cite>
                  <time className="review__date" dateTime={comment.date.dateTime}>
                    {comment.date.humanizeDate}
                  </time>
                </footer>
              </blockquote>

              <div className="review__rating">{comment.rating}</div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

FilmPageReviews.propTypes = {
  backgroundColor: PropTypes.string.isRequired,
  comments: PropTypes.arrayOf(PropTypes.shape({
    user: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.shape({
      humanizeDate: PropTypes.string.isRequired,
      dateTime: PropTypes.string.isRequired,
    }).isRequired,
  })).isRequired,
  filmId: PropTypes.number.isRequired,
  loadComments: PropTypes.func.isRequired,
};

const mapStateToProps = ({APP_STATE}) => ({
  comments: APP_STATE.comments,
});

const mapDispatchToProps = (dispatch) => ({
  loadComments(id) {
    dispatch(fetchComments(id));
  }
});

export {FilmPageReviews};
export default connect(mapStateToProps, mapDispatchToProps)(FilmPageReviews);
