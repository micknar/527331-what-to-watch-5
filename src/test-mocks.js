import {getDate} from "./utils";

export const noop = () => { };

export const films = [
  {
    id: 1,
    name: `Bronson`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `/img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
    rating: 8.9,
    ratingMark: `Booring`,
    scoresCount: 240,
    director: `Nicolas Winding Refn`,
    starring: [`Bill Murray`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 40,
    },
    genre: `Crime`,
    released: 1996,
    isFavorite: false,
  },
  {
    id: 2,
    name: `Midnight Special`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Midnight_Special.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 10,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 42,
    },
    genre: `Thriller`,
    released: 1989,
    isFavorite: true,
  },
  {
    id: 3,
    name: `Pulp Fiction`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Pulp_Fiction.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amo eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 2.2,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 22,
    },
    genre: `Action`,
    released: 2007,
    isFavorite: false,
  },
  {
    id: 4,
    name: `Matrix`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Matrix.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ext enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.5,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 24,
    },
    genre: `Adventure`,
    released: 2011,
    isFavorite: true,
  },
  {
    id: 5,
    name: `Legend`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Legend.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem i ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 4.1,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 57,
    },
    genre: `Comedy`,
    released: 2015,
    isFavorite: false,
  },
  {
    id: 6,
    name: `angs of new york`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/gangs_of_new_york.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sedrem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.uip ex ea commodo consequat.`,
    rating: 7.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 35,
    },
    genre: `Fantasy`,
    released: 1980,
    isFavorite: false,
  },
  {
    id: 7,
    name: `A Star Is Born`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_is_Born.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_is_Born.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 6.4,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Guy Ritchie`,
    starring: [`Bill Murray`, `Edward Norton`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 1,
      minutes: 25,
    },
    genre: `Drama`,
    released: 1989,
    isFavorite: false,
  },
  {
    id: 8,
    name: `Snatch`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Snatch.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Snatch.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliqliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`,
    rating: 9.8,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Edward Norton`, `Edward Norton`, `Edward Norton`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: {
      hours: 2,
      minutes: 20,
    },
    genre: `Fantasy`,
    released: 2004,
    isFavorite: false,
  },
  {
    id: 9,
    name: `Bohemian Rhapsody`,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Bohemian_Rhapsody.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/Bohemian_Rhapsody.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Bohemian_Rhapsody.jpg`,
    backgroundColor: `#4aa2e5`,
    videoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `Lorem ipsum dolor sit amet, consectetur.`,
    rating: 3,
    ratingMark: `Booring`,
    scoresCount: `240`,
    director: `Quentin Tarantino`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`],
    runTime: {
      hours: 1,
      minutes: 20,
    },
    genre: `Adventure`,
    released: 1998,
    isFavorite: false,
  },
];

export const comments = [
  {
    user: {
      id: 19,
      name: `Christina`,
    },
    rating: 2.4,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 3.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-27T13:38:44.769Z`),
  },
  {
    user: {
      id: 2,
      name: `Mollie`,
    },
    rating: 1.7,
    comment: `Lorem ipsustrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-09T13:38:44.769Z`),
  },
  {
    user: {
      id: 14,
      name: `Corey`,
    },
    rating: 4.2,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ilaboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-11T13:38:44.769Z`),
  },
  {
    user: {
      id: 11,
      name: `Jack`,
    },
    rating: 4.5,
    comment: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitatboris nisi ut aliquip ex ea commodo consequat`,
    date: getDate(`2020-10-19T13:38:44.769Z`),
  },
];
