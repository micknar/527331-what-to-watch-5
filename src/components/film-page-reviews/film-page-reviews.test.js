import React from "react";
import renderer from "react-test-renderer";
import {FilmPageReviews} from "./film-page-reviews";
import {noop, comments} from "../../test-mocks";

it(`Should FilmPageReviews render correctly`, () => {
  const tree = renderer
    .create(
        <FilmPageReviews
          comments={comments}
          filmId={1}
          loadComments={noop}
          backgroundColor={`#4aa2e5`}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
