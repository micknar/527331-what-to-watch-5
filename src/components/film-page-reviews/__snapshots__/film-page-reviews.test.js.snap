// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmPageReviews render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
      style={
        Object {
          "borderBottomColor": "#3b81b7",
        }
      }
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Christina
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-9"
          >
            October 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        2.4
      </div>
    </div>
    <div
      className="review"
      style={
        Object {
          "borderBottomColor": "#3b81b7",
        }
      }
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Corey
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-27"
          >
            October 27, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.2
      </div>
    </div>
    <div
      className="review"
      style={
        Object {
          "borderBottomColor": "#3b81b7",
        }
      }
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsustrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Mollie
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-9"
          >
            October 9, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        1.7
      </div>
    </div>
    <div
      className="review"
      style={
        Object {
          "borderBottomColor": "#3b81b7",
        }
      }
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ilaboris nisi ut aliquip ex ea commodo consequat
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Corey
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-11"
          >
            October 11, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        4.2
      </div>
    </div>
    <div
      className="review"
      style={
        Object {
          "borderBottomColor": "#3b81b7",
        }
      }
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitatboris nisi ut aliquip ex ea commodo consequat
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Jack
          </cite>
          <time
            className="review__date"
            dateTime="2020-10-19"
          >
            October 19, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        4.5
      </div>
    </div>
  </div>
</div>
`;
